### ISSUE TITLE
Cannot set up Kraken API if Kraken has no balances

### ISSUE THREAD

**Issue by aspiers** [rotki/rotki#2443]
## Problem Definition

It fails every time I try to add my Kraken API keys:

![kraken](https://user-images.githubusercontent.com/100738/109203071-9f3ad000-779b-11eb-929e-8031f293ef97.png) 

The keys are definitely correct ;-)  I have tried this several times.

## Logs

```
[25/02/2021 18:07:46 GMT] DEBUG rotkehlchen.exchanges.manager: Initializing exchanges
[25/02/2021 18:07:46 GMT] INFO rotkehlchen.exchanges.exchange: Initialized kraken exchange
[25/02/2021 18:07:46 GMT] DEBUG rotkehlchen.exchanges.kraken: Kraken API query method=Balance, data=None, call_counter=0
[25/02/2021 18:07:47 GMT] DEBUG rotkehlchen.exchanges.coinbasepro: Got report for id: 90d122b0-a7ca-4b3a-ad3a-c2127f14003e with length 95. Skipping it
[25/02/2021 18:07:47 GMT] DEBUG rotkehlchen.exchanges.coinbasepro: Coinbase Pro API query request_method=GET, request_url=/reports/6c8980e1-fd65-47b9-8318-b7023118171b, options={}
[25/02/2021 18:07:47 GMT] ERROR rotkehlchen.exchanges.kraken: Kraken API key validation error: Unexpected format of Kraken response. Missing key: 'result'
[25/02/2021 18:07:47 GMT] ERROR rotkehlchen.exchanges.manager: Failed to validate API key for exchange name=kraken, error=Unknown error at Kraken API key validation. Perhaps API Key/Secret combination invalid?
[25/02/2021 18:07:47 GMT] DEBUG rotkehlchen.api.rest: Request successful response={"result": null, "message": "Unknown error at Kraken API key validation. Perhaps API Key/Secret combination invalid?"}, status_code=409
```

### System Description

Operating system: openSUSE Tumbleweed
Rotki version: git `develop` branch, previously seen with older AppImage releases too

**Comment by aspiers** [comment]
Previously discussed [here](https://discord.com/channels/657906918408585217/745172610240872508/798616122337001504) and [here](https://discord.com/channels/657906918408585217/745172610240872508/814565875147341865).

**Comment by LefterisJP** [comment]
Do you have zero balances in there? I recall there being a funny problem with this and kraken api response. Something about not following their docs. Essentially as you see there. Missing the actual "result" from the response if there is no balances.

![2021-02-25_20-34](https://user-images.githubusercontent.com/1658405/109206976-d95e9e80-77a8-11eb-9686-3963f2a82130.png)

The chat is with a Kraken manager back in May 2020. We fixed it back then. But by checking the code I don't see it anymore. So seems like we introduced a regression ... and they never fixed it on their side. If you can confirm that you got no balances then this is it.

**Comment by aspiers** [comment]
Ah! Nice catch. Yes I have zero balances currently.

**Comment by LefterisJP** [comment]
Should be fixed in bugfixes branch thanks to #2445
